src/coreclr/vm/object.cpp

```diff
@@ -1394,6 +1394,24 @@ void __fastcall ZeroMemoryInGCHeap(void* mem, size_t size)
         *memBytes++ = 0;
 }
 
+void DelegateObject::SetMethodBase(OBJECTREF newMethodBase)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+    
+    GCX_COOP();
```

Comment:
```suggestion
        MODE_COOPERATIVE;
    }
    CONTRACTL_END;
    
```
Methods that take OBJECTREF arguments must run in cooperative mode.